server/athenian/api/connexion.py

```diff
@@ -298,17 +299,24 @@ async def connect_to_db(name: str, shortcut: str, db_conn: str, db_options: dict
 
     def on_dbs_conected(self, callback):
         """Register a callback on dbs connected."""
+        ev = asyncio.Event()
+        self._on_dbs_conected_events.append(ev)
 
         def callback_wrapper(*_):
             dbs = {
                 db: getattr(self, db)
                 for db in self._db_futures
             }
             callback(**dbs)
+            ev.set()
```

Comment:
The problem is that here I actually want not only to wait for the dbs connected, but also on the callback being finished. AFAIK there's no other way to `await` for the callback since it's not an `async` task.

Reply:
I mean this:

```python
async def task_wrapper():
    await gather(*self._db_futures.values())
    dbs = {
        db: getattr(self, db)
        for db in self._db_futures
    }
    callback(**dbs)
asyncio.create_task(task_wrapper(), name=f"on_dbs_conected({callback})")
```

Reply:
done, I needed to use `ensure_future` since there was no loop running yet though.